File: csharp/tests/Printers.cs
using System.Collections.Generic;
using NecklaceRefactoringKata.JewelleryTypes;

namespace NecklaceRefactoringKata.Tests;

public class Printers
{
    public static string PrintJewelleryStorage(JewelleryStorage storage)
    {
        return $@"Jewellery Storage:
Box:
    Ring Compartment: {PrintList(storage.Box.RingCompartment)}
    Top Shelf:        {PrintList(storage.Box.TopShelf)}
    Main Section:     {PrintList(storage.Box.MainSection)}
Tree:                 {PrintList(storage.Tree)}
Travel Roll:          {PrintList(storage.TravelRoll)}
Safe:                 {PrintList(storage.Safe)}
On top of dresser:    {PrintList(storage.DresserTop)}
";
    }
    public static string PrintList(List<JewelleryBase> items)
    {
        return string.Join(", ", items);
    }
}

File: java/src/test/java/necklaceRefactoringKata/jewellery/Printers.java
package necklaceRefactoringKata.jewellery;

import necklaceRefactoringKata.JewelleryStorage;
import necklaceRefactoringKata.jewellery.Jewellery;
import necklaceRefactoringKata.jewellery.Ring;

import java.util.List;

public class Printers {
    public static String PrintJewelleryStorage(JewelleryStorage storage)
    {
        return "Jewellery Storage: " + "\n" +
        "Box:" + "\n" +
        "    Ring Compartment: " + PrintRings(storage.box.ringCompartment) + "\n" +
        "    Top Shelf:        " + PrintList(storage.box.topShelf) + "\n" +
        "    Main Section:     " + PrintList(storage.box.mainSection) + "\n" +
        "Tree:                 " + PrintList(storage.tree) + "\n" +
        "Travel Roll:          " + PrintList(storage.travelRoll) + "\n" +
        "Safe:                 " + PrintList(storage.safe) + "\n" +
        "On top of dresser:    " + PrintList(storage.dresserTop) + "\n"
        ;
    }
    public static String PrintRings(List<Ring> items)
    {
        return items.toString();
    }
    public static String PrintList(List<Jewellery> items)
    {
        return items.toString();
    }

}
